### PR TITLE
Refactoring runner resource container linedelimiter to it's own pkg

### PR DESCRIPTION
**Pull request by resouer**
Continuing my work ref #15634

Anyone is ok to review this fix.

### CODE REVIEW

**Comment by resouer** [comment]
@k8s-bot e2e test this please ref #23545

**Comment by resouer** [comment]
@k8s-bot test this please ref #23545

**Comment by resouer** [comment]
ping @thockin , submit queue is back, please review.

**Comment by thockin** [comment]
LGTM.  Not critical for 1.3, though.

**Comment by resouer** [comment]
@k8s-bot unit test this issue: #27179

**Comment by resouer** [comment]
@thockin What is merge rule for `Milestone is for a future release and cannot be merged`, after 1.3?

**Comment by fejta** [comment]
@k8s-bot node e2e test this issue: #28047

File: pkg/util/linedelimiter/line_delimiter.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package util
+package linedelimiter
```
Comment (thockin) [inline pkg/util/linedelimiter/line_delimiter.go]: This could go in util/strings couldn't it?

File: cmd/kubelet/app/server.go [hunk 2]
```diff
@@ -674,7 +674,7 @@ func RunKubelet(kcfg *KubeletConfig) error {
 		return fmt.Errorf("failed to create kubelet: %v", err)
 	}
 
-	util.ApplyRLimitForSelf(kcfg.MaxOpenFiles)
+	resourcecontainer.ApplyRLimitForSelf(kcfg.MaxOpenFiles)
```
Comment (thockin) [inline cmd/kubelet/app/server.go]: This function is very poorly named, and it is used in exactly one place - kubelet.  Can we just move it to kubelet and call it rlimitNumFiles()

File: pkg/util/runner/runner.go [hunk 3]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package util
+package runner
```
Comment (thockin) [inline pkg/util/runner/runner.go]: how about util/async

File: cmd/kubelet/app/server.go (hunk 2 above)
Comment (thockin) [inline cmd/kubelet/app/server.go]: or util/os

Comment (thockin) [inline cmd/kubelet/app/server.go]: or util/os/rlimit or even util/rlimit

Comment (resouer) [inline cmd/kubelet/app/server.go]: I think this can be moved into kubelet, will refactor this

Comment (resouer) [inline cmd/kubelet/app/server.go]: I changed my idea, will move it into util/rlimit

**Comment by resouer** [comment]
@thockin Fixed and rebased, please review

File: pkg/util/async/runner.go [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2014 The Kubernetes Authors.
+Copyright 2014 The Kubernetes Authors All rights reserved.
```
Comment (thockin) [inline pkg/util/async/runner.go]: You'll need to remove the "All rights reserved" from these - new boilerplate has removed it.

**Comment by thockin** [comment]
LGTM except jenkins (probably the "all rights reserved".  Thanks!  Sometimes less new packages is better :)

**Comment by thockin** [comment]
ping me when CI is green, please

**Comment by resouer** [comment]
@thockin Everything is ok :)

**Comment by resouer** [comment]
ping @thockin Everything is ok :)

**Comment by resouer** [comment]
ping @thockin

**Comment by resouer** [comment]
ping @thockin, this is ready to be merged

**Comment by resouer** [comment]
@thockin Fixed error caused by rebase again, please set label if looks good.

**Comment by resouer** [comment]
kindly ping @thockin, please set label :)

**Comment by thockin** [comment]
LGTM

**Pull request merged**